src/util/Strudel.re

```diff
@@ -0,0 +1,67 @@
+open Js_of_ocaml;
+
+/* Direct bindings to Strudel functions */
+let initStrudel: unit => unit =
+  () => {
+    let initStrudelFn = Js.Unsafe.js_expr("window.initStrudel");
+    Js.Unsafe.fun_call(initStrudelFn, [||]);
+  };
+
+let hush: unit => unit =
+  () => {
+    let hushFn = Js.Unsafe.js_expr("window.hush");
+    Js.Unsafe.fun_call(hushFn, [||]);
+  };
+
+let note: string => Js.Unsafe.any =
+  pattern => {
+    let noteFn = Js.Unsafe.js_expr("window.note");
+    Js.Unsafe.fun_call(noteFn, [|Js.Unsafe.inject(Js.string(pattern))|]);
+  };
+
+let rev: Js.Unsafe.any => Js.Unsafe.any =
+  pattern => {
+    let revFn = Js.Unsafe.js_expr("window.rev");
+    Js.Unsafe.fun_call(revFn, [|Js.Unsafe.inject(pattern)|]);
+  };
+
+let jux: (Js.Unsafe.any, Js.Unsafe.any => Js.Unsafe.any) => Js.Unsafe.any =
+  (pattern, f) => {
+    Js.Unsafe.meth_call(pattern, "jux", [|Js.Unsafe.inject(f)|]);
+  };
+
+let play: Js.Unsafe.any => unit =
+  pattern => {
+    Js.Unsafe.meth_call(pattern, "play", [||]);
+  };
+
+/* Wrapper function to chain methods */
+let playNote: string => unit =
+  pattern => {
+    let n = note(pattern);
+    let j = jux(n, rev);
+    play(j);
+  };
+
+/* Example usage function */
+let exampleUse: unit => unit =
+  () => {
+    initStrudel();
+    playNote("<c a f e>(3,8)");
+  };
+
+/* Function to stop the music */
+let stopMusic: unit => unit = () => hush();
+
+/* Function to initialize Strudel when the DOM is loaded */
+let initOnLoad: unit => unit =
+  () => {
+    let addEventListenerFn = Js.Unsafe.js_expr("window.addEventListener");
+    Js.Unsafe.fun_call(
+      addEventListenerFn,
+      [|
+        Js.Unsafe.inject(Js.string("DOMContentLoaded")),
+        Js.Unsafe.inject(Js.wrap_callback(_ => initStrudel())),
+      |],
+    );
+  };
```

Comment:
```suggestion
let initStrudel: unit => unit =
  () => Js.Unsafe.coerce(Dom_html.window)##initStrudel();

let hush: unit => unit =
  () => {
    Js.Unsafe.coerce(Dom_html.window)##hush();
  };

let note: string => Js.Unsafe.any =
  pattern => {
    Js.Unsafe.coerce(Dom_html.window)##note(Js.string(pattern));
  };

let rev: Js.Unsafe.any => Js.Unsafe.any =
  pattern => {
    Js.Unsafe.coerce(Dom_html.window)##rev(pattern);
  };

let jux: (Js.Unsafe.any, Js.Unsafe.any => Js.Unsafe.any) => Js.Unsafe.any =
  (pattern, f) => {
    Js.Unsafe.coerce(pattern)##jux(f);
  };

let play: Js.Unsafe.any => unit =
  pattern => {
    Js.Unsafe.coerce(pattern)##play();
  };

/* Wrapper function to chain methods */
let playNote: string => unit =
  pattern => {
    let n = note(pattern);
    let j = jux(n, rev);
    play(j);
  };

/* Example usage function */
let exampleUse: unit => unit =
  () => {
    initStrudel();
    playNote("<c a f e>(3,8)");
  };

/* Function to stop the music */
let stopMusic: unit => unit = () => hush();

/* Function to initialize Strudel when the DOM is loaded */
let initOnLoad: unit => Dom_events.listener =
  () => {
    Dom_events.listen(
      Dom_html.window,
      Dom_events.Typ.domContentLoaded,
      (_: Js.t(Dom_html.window), _: Js.t(Dom_html.event)) => {
        initStrudel();
        false; // I don't think this does anything. 
      },
    );
  };
```

Reply:
The `false` being returned on the `Dom_events.listen` looks like it would be used for stop propagation but I looked through the js_of_ocaml implementation and I don't see where this is happening.

Reply:
I take it back https://github.com/ocsigen/js_of_ocaml/blob/ea51f5e6e1c54a13eb917d081c51175814b8b58b/lib/js_of_ocaml/dom_events.ml#L32 the newest version calls full_handler which does prevent_default so probably set that to `true`